src/vl-map-search.js

```diff
@@ -146,22 +174,40 @@ export class VlMapSearch extends vlElement(HTMLElement) {
       this.__choiceEventListenerRegistered = true;
       this.__choiceEventListener = this._selectElement.addEventListener('choice', (event) => {
         if (event && event.detail && event.detail.choice) {
-          fetch(this.locationUrl + event.detail.choice.value).then((response) => {
-            return response.json();
-          }).then((data) => {
-            if (data && data.LocationResult) {
-              if (this._onSelect) {
-                this._onSelect(data);
-              } else if (this._map) {
-                this._map.zoomTo([
-                  data.LocationResult[0].BoundingBox.LowerLeft.X_Lambert72,
-                  data.LocationResult[0].BoundingBox.LowerLeft.Y_Lambert72,
-                  data.LocationResult[0].BoundingBox.UpperRight.X_Lambert72,
-                  data.LocationResult[0].BoundingBox.UpperRight.Y_Lambert72,
-                ]);
+          const value = event.detail.choice.value;
+
+          // TODO stefanborghys: 28/01/21 prefer to use instanceof but could not import LambertCoordinaat in the test and replaced it for now
+          // instanceof LambertCoordinaat
+          if (value.x !== undefined && value.y !== undefined ) {
+            this._map.zoomToGeometry({
+              type: 'Point',
+              coordinates: [value.x, value.y],
+            }, 10);
+          } else if (value instanceof Object) {
+            this._map.zoomTo([
+              value.BoundingBox.LowerLeft.X_Lambert72,
+              value.BoundingBox.LowerLeft.Y_Lambert72,
+              value.BoundingBox.UpperRight.X_Lambert72,
+              value.BoundingBox.UpperRight.Y_Lambert72,
+            ], 14);
+          } else {
+            fetch(this.locationUrl + value).then((response) => {
+              return response.json();
+            }).then((data) => {
+              if (data && data.LocationResult) {
```

Comment:
misschien ook nog op de length van LocationResult testen?

Reply:
waarom eigenlijk, ik dacht dat je hier altijd met zekerheid een resultaat kan verwachten want je geeft hier dezelfde keuze mee als wat je uit de suggestie hebt gekregen bij het zoeken. Zou dus denken dat wanneer je bij geopunt een suggestie krijgt deze daarna nog steeds als locatie op te vragen is met diezelfde API. Nu het is geen argument maar deze code stond er ook al zo waardoor ik er me geen verdere vragen heb over gesteld.